modules/st2-auto-form/fields/array.js

```diff
@@ -71,14 +71,18 @@ export default class ArrayField extends BaseTextField {
   }
 
   toStateValue(v) {
-    if (jsonCheck(v) || Array.isArray(v)) {
+    if (jsonCheck(v)) {
       return JSON.stringify(v);
     }
 
     if (isJinja(v)) {
       return v;
     }
 
+    if (Array.isArray(v) && this.props.spec.items.type === 'object') {
```

Comment:
This is the offending line in the scenario where actions may not above spec.items defined

Reply:
You guys are right, I did not test with the yaml definition not having the type of item. Let me take another look on how I can distinguish what type of items are in the array.